cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -40,37 +42,57 @@ class parquet_field {
   int field() const { return _field_val; }
 };
 
+std::string field_type_string(FieldType type)
+{
+  switch (type) {
+    case ST_FLD_TRUE: return "bool(true)";
+    case ST_FLD_FALSE: return "bool(false)";
+    case ST_FLD_BYTE: return "int8";
+    case ST_FLD_I16: return "int16";
+    case ST_FLD_I32: return "int32";
+    case ST_FLD_I64: return "int64";
+    case ST_FLD_DOUBLE: return "double";
+    case ST_FLD_BINARY: return "binary";
+    case ST_FLD_STRUCT: return "struct";
+    case ST_FLD_LIST: return "list";
+    case ST_FLD_SET: return "set";
+    default: return "unknown(" + std::to_string(type) + ")";
+  }
+}
+
+void assert_field_type(int type, FieldType expected)
+{
+  CUDF_EXPECTS(type == expected,
```

Comment:
Oh no, this is comparing `int` vs `enum`.

Reply:
We always compared `int`s and `FieldType` in this code, e.g. `if (field_type != ST_FLD_LIST) { return true; }`. I think it's fine, given that we throw if the values are not equal - no int values that are not valid `FieldType` values will get past this check. We have to sanitize the input somewhere :man_shrugging:

Reply:
We definitely need to refactor it, changing into `enum class` for `FieldType`.

Reply:
I'm not following. We have to compare the int input with a FieldType value somewhere. Using an enum class helps with the scope of the value names, but we still need the int <-> enum comparison, just with a `static_cast`.

Reply:
Then we will do an explicit `static_cast` before comparison. I think it is better than this implicit comparison.

Reply:
Just checked, `FieldType` is not exactly part of the spec, we have some creative freedom around it. 
If you don't mind, let's make it an `enum class` in a separate PR. It's a wider change than the error handling this PR is about.